asb-native/src/main/java/org/ballerinax/asb/sender/MessageSender.java

```diff
@@ -180,13 +183,84 @@ public Object sendBatch(BMap<BString, Object> messages) {
+        byte[] byteArray;
+        Type type = TypeUtils.getType(messageBody);
+        if (type.getTag() == TypeTags.STRING_TAG) {
+            byteArray = ((BString) messageBody).toString().getBytes();
+        } else if (type.getTag() == TypeTags.INT_TAG) {
+            byteArray = Integer.toString((int) messageBody).getBytes();
+        } else {
+            byteArray = ((BArray) messageBody).getBytes();
+        }
+
+        ServiceBusMessage asbMessage = new ServiceBusMessage(byteArray);
+
+        if (message.containsKey(StringUtils.fromString(ASBConstants.CONTENT_TYPE))) {
+            String contentType = message.getStringValue(StringUtils.fromString(ASBConstants.CONTENT_TYPE)).getValue();
+            asbMessage.setContentType(contentType);
+        }
+        if (message.containsKey(StringUtils.fromString(ASBConstants.MESSAGE_ID))) {
+            String messageId = message.getStringValue(StringUtils.fromString(ASBConstants.MESSAGE_ID)).getValue();
+            asbMessage.setMessageId(messageId);
+        }
+        if (message.containsKey(StringUtils.fromString(ASBConstants.TO))) {
+            String to = message.getStringValue(StringUtils.fromString(ASBConstants.TO)).getValue();
+            asbMessage.setTo(to);
+        }
+        if (message.containsKey(StringUtils.fromString(ASBConstants.REPLY_TO))) {
+            String replyTo = message.getStringValue(StringUtils.fromString(ASBConstants.REPLY_TO)).getValue();
+            asbMessage.setReplyTo(replyTo);
+        }
+        if (message.containsKey(StringUtils.fromString(ASBConstants.REPLY_TO_SESSION_ID))) {
+            String replyToSessionId = message.getStringValue(StringUtils.fromString(ASBConstants.REPLY_TO_SESSION_ID))
+                    .getValue();
+            asbMessage.setReplyToSessionId(replyToSessionId);
+        }
+        if (message.containsKey(StringUtils.fromString(ASBConstants.LABEL))) {
+            String subject = message.getStringValue(StringUtils.fromString(ASBConstants.LABEL)).getValue();
+            asbMessage.setSubject(subject);
+        }
+        if (message.containsKey(StringUtils.fromString(ASBConstants.SESSION_ID))) {
+            String sessionId = message.getStringValue(StringUtils.fromString(ASBConstants.SESSION_ID)).getValue();
+            asbMessage.setSessionId(sessionId);
+        }
+        if (message.containsKey(StringUtils.fromString(ASBConstants.CORRELATION_ID))) {
+            String correlationId = message.getStringValue(StringUtils.fromString(ASBConstants.CORRELATION_ID))
+                    .getValue();
+            asbMessage.setCorrelationId(correlationId);
+        }
+        if (message.containsKey(StringUtils.fromString(ASBConstants.PARTITION_KEY))) {
+            String partitionKey = message.getStringValue(StringUtils.fromString(ASBConstants.PARTITION_KEY)).getValue();
+            asbMessage.setPartitionKey(partitionKey);
+        }
+
```

Comment:
Need to format the file here. Whitespaces are inconsistent.

Reply:
Ack. Will fix the formatting issues.